hist/hist/src/TAxis.cxx

```diff
@@ -311,7 +311,7 @@ Int_t TAxis::FindBin(Double_t x)
       return FindFixBin(x);
    } else {
       if (!fXbins.fN) {        //*-* fix bins
-         bin = 1 + int (fNbins*(x-fXmin)/(fXmax-fXmin) );
+         bin = 1 + round( (double)fNbins*((x-fXmin)/(fXmax-fXmin)) );
```

Comment:
I don't think this is correct. 
`std::round` approximate to the closes integer and so if x is closer to the upper edge it will move to the bin+1. For example for `xmin=0, xmax=10,nbins=10` you have for `x=0.6` `bin=1+1=2` instead of the correct value of 1

Reply:
And I think the operation is done anyway in double precision so I am not sure if adding `(double)` is going to be helpful. 
How is it called `FindFixBin` when setting the log scale ?

Reply:
int does not work. The reproducer is the macro I posted in this PR. I agree that can be questionable but, I spend a lot of time on it. Try the macro and you will see.